src/llm_tracekit/bedrock/invoke_agent.py

```diff
@@ -102,38 +103,51 @@ def record_invoke_agent_result_attributes(
             tool_calls=tool_calls,
         )
 
-        if result.prompt_history is not None:
-            all_prompts = result.prompt_history
-        else:
-            all_prompts = []
+        final_attributes = {}
 
-        if result.completion_history is not None:
+        if result.prompt_history is not None and result.completion_history is not None:
+            all_prompts = result.prompt_history
             all_choices = result.completion_history + [current_choice]
+
+            final_attributes.update(
+                generate_message_attributes(
+                    messages=all_prompts, capture_content=capture_content
+                )
+            )
+            final_attributes.update(
+                generate_choice_attributes(
+                    choices=all_choices,
+                    capture_content=capture_content,
+                )
+            )
```

Comment:
shouldn't these be two separate conditions? `if result.prompt_history is not None` -> record message attributes, and `if result.completion_history is not None` -> record choice attributes